plotextractor/converter.py

```diff
@@ -22,45 +22,42 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
+"""Functions related to conversion and untarring."""
+
+from __future__ import absolute_import, print_function, unicode_literals
+
 import os
 import tarfile
+import re
 
-from invenio.utils.shell import run_shell_command, run_process_with_timeout, Timeout
-from .output_utils import get_converted_image_name, \
-                                               write_message
+from subprocess32 import check_output, TimeoutExpired
+from wand.image import Image
 
-def untar(original_tarball, sdir):
-    """
-    Here we decide if our file is actually a tarball (sometimes the
-    'tarballs' gotten from arXiv aren't actually tarballs.  If they
-    'contain' only the TeX file, then they are just that file.), then
-    we untar it if so and decide which of its constituents are the
-    TeX file and which are the images.
+from .errors import InvalidTarball
+from .output_utils import get_converted_image_name, get_image_location
 
-    @param: tarball (string): the name of the tar file from arXiv
-    @param: dir (string): the directory where we would like it untarred to
 
-    @return: (image_list, tex_file) (([string, string, ...], string)):
-        list of images in the tarball and the name of the TeX file in the
-        tarball.
-    """
+def untar(original_tarball, output_directory=None):
+    """Untar given tarball file into directory.
 
-    if not tarfile.is_tarfile(original_tarball):
-        return ([], [], None)
+    Here we decide if our file is actually a tarball, then
+    we untar it and return a list of extracted files.
 
-    tarball = tarfile.open(original_tarball)
-    tarball.extractall(sdir)
+    :param: tarball (string): the name of the tar file from arXiv
+    :param: output_directory (string): the directory to untar in
 
-    tex_output_contains = 'TeX'
+    :return: list of absolute file paths
+    """
+    if not tarfile.is_tarfile(original_tarball):
+        raise InvalidTarball
 
-    tex_file_extension = 'tex'
-    image_output_contains = 'image'
-    eps_output_contains = '- type eps'
-    ps_output_contains = 'Postscript'
+    tarball = tarfile.open(original_tarball)
+    if not output_directory:
+        # No directory given, so we use the same directory as the tarball
+        output_directory = os.path.dirname(original_tarball)
```

Comment:
At least be sure the checkmanifest trick is in place, which can save your day when making a release later on pypi